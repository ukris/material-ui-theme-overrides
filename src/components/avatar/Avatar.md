Avatar Standard Example:
```jsx
    <Avatar
       
    />
```

Avatar with Size and shape and invalid image

```jsx
    <Avatar
        src="https://www.gravatarz.com/avatar/fooff"
        name="Natasha Special"
        size="xl"
        shape="round"
        border
    />
```

