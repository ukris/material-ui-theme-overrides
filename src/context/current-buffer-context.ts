import { createContext } from 'react'
import { Buffer } from 'types'

interface ICurrentBufferContextProps {
    currentBuffer?: Buffer[]
}

const CurrentBufferContext  = createContext({} as ICurrentBufferContextProps)

export default CurrentBufferContext