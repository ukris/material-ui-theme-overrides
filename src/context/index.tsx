import CurrentBufferContext from './current-buffer-context'
import TableContext from './table-context'
import GlobalContext from './global-context'
import ProjectContext from './project-context'

export { CurrentBufferContext,  GlobalContext, ProjectContext, TableContext }