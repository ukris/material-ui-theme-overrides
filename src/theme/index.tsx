import { themePalette, LIGHT, DARK } from './palette'
export type LIGHTORDARK = 'light' | 'dark'
export { default } from './theme-provider'
export { default as theme } from './theme'
export { default as makePalette,  } from './palette'
export { default as typography } from './typography'
export { default as ThemeContext } from './theme-context'
export { default as ThemeProvider } from './theme-provider'
export { themePalette, LIGHT, DARK }

const Themes = {
  basil: null,
  crane: null,
  material: null,
  materialDark: null,
  pinky: null,
  rally: null,
  reply: null,
}
export type Color = 'primary' | 'success' | 'info' | 'accent' | 'warning' | 'danger' | 'error' | 'default'
export type ColorValue = {
    light: string,
    main: string,
    dark: string,
    contrastText: string
}
export type ColorObj = {
  [key in Color] : ColorValue
}

export type ThemeType = keyof typeof Themes
export type ThemeZoom = {
  theme: ThemeType
  zoom: number
}
export const AVAILABLE_THEMES = Object.keys(Themes) as ThemeType[]
export const unit = 'rem'
export const tablePadding = [1,0]
export const FONT_BASIS = 2