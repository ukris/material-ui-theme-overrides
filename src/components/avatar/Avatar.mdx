---
name: Ava
route: /
---

# Ava

Avatar

import Avatar from "./Avatar";
import {Playground} from 'docz'

### Avatar

<Playground>
  <Avatar/>
  <Avatar
      src="https://www.gravatarz.com/avatar/fooff"
      name="Natasha Special"
      size="xl"
      shape="round"
      border
  />
</Playground>