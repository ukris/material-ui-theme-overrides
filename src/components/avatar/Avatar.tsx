import React, { ReactNode, useState, useEffect } from 'react'
import clsx from 'clsx'

import Person from '@material-ui/icons/Person'
import { OverridableComponent } from '@material-ui/core/OverridableComponent'
import { SvgIconTypeMap } from '@material-ui/core/SvgIcon'
import { WithStyles, withStyles, Theme} from '@material-ui/core/styles'

import { hashNumber, getInitials } from 'helpers/misc'
import GlobalContext from 'context/global-context'
import { Size} from 'types'

import { LIGHT, WHITE, BLACK, getHashColor } from 'theme/palette'
import { IAvatar, Shape } from './types'

// @ts-ignore
export function valToAvatars(values:any[], col?:any):IAvatar[]  {
  return (values||[]).map(val=>{
    return{
      src:val[0],
      name: val[1]
    }
  })
}

const styles = (theme: Theme):any => {
    const { fontSize, palette, shape, zoomSpacing,typography } = theme
    const color =  palette.type === 'light' ? palette.grey[400] : palette.grey['A700']
    const def = palette.background.default
    const spacing2 = zoomSpacing(2)
    const spacing3 = zoomSpacing(3)
    const spacing4 = zoomSpacing(4)
    const spacing5 = zoomSpacing(5)
    const spacing6 = zoomSpacing(6)

    return {
        root: {
            position: 'relative',
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            flexShrink: 0,
            fontFamily: typography.fontFamily,
            lineHeight: 1,
            overflow: 'hidden',
            userSelect: 'none',
        },
        /* Styles applied to the root element if not `src` or `srcSet`. */
        colorDefault: {
            color: def,
            backgroundColor: color
        },
        /* Styles applied to the root element if `variant="circle"`. */
        circle: {
          borderRadius: '50%',
        },
        border: {
          border: `2px solid ${def}`
        },
        noClass: {
          
        },
        /* Styles applied to the root element if `variant="rounded"`. */
        round: {
            borderRadius: shape.borderRadius,
        },
        /* Styles applied to the root element if `variant="square"`. */
        square: {
            borderRadius: 0,
        },
        /* Styles applied to the img element if either `src` or `srcSet` is defined. */
        img: {
            width: '100%',
            height: '100%',
            // Handle non-square image. The property isn't supported by IE 11.
            objectFit: 'contain',
            // Hide alt text.
            color: 'transparent',
        },
        xs: {
            width: spacing2,
            height: spacing2,
            fontSize: fontSize.xs
        },
        sm: {
            width: spacing3,
            height: spacing3,
            fontSize: fontSize.sm
        },
        md: {
            width: spacing4,
            height: spacing4,
            fontSize: fontSize.md
        },
        lg: {
            width:  spacing5,
            height: spacing5,
            fontSize: fontSize.lg
        },
        xl: {
            width: spacing6,
            height: spacing6,
            fontSize: fontSize.xl
        }
    }
}

export interface StyleProps  extends WithStyles<typeof styles> {
   
}

export interface AvatarProps extends StyleProps, IAvatar{
    imgProps?: React.ImgHTMLAttributes<HTMLImageElement>
    size?: Size
    shape?: Shape
    border?: boolean
    className?: string
    style?: object
    children?: ReactNode
    component?: any
    custom?: boolean
    extra?: boolean
    initials?: number
    cStyle?: object
    Icon?:  OverridableComponent<SvgIconTypeMap<{}, "svg">> | null
}

const Avatar = function Avatar(props: AvatarProps) {
    
    const {
      name,
      children: childrenProp,
      classes,
      className,
      component: Component = 'div',
      imgProps,
      src,
      custom,
      size='md',
      shape = 'circle',
      border = true,
      style={},
      extra,
      cStyle={},
      Icon,
      handleClick,
      initials=2,
      ...other
    } = props
  
    if (handleClick) {
      // @ts-ignore
      style.cursor = 'pointer'
    }
    const { palette } = GlobalContext.theme
    const [loadedSrc, setLoadedSrc] = useState<boolean>(false)

    let children = null;
    const borderCls = border ? 'border' : 'noClass'

    useEffect(()=> {
      if (src) {
        const image = new Image()
        image.src = src
        image.onload = () => {
          setLoadedSrc(true)
        }
        image.onerror = () => {
        }
      }
    },[src, setLoadedSrc ])

    if (loadedSrc) {
      children = (
        <img
          alt={name}
          src={src} 
          className={classes.img}
          {...imgProps}
        />
      )
    } else if (childrenProp != null) {
      children = childrenProp;
    } else if (name) {
      if (!extra) {
        const number = hashNumber(name)
        const color = getHashColor(number)
        children = getInitials(name,initials)
        // @ts-ignore
        style.backgroundColor = color
          // @ts-ignore
        if (!palette) {
          debugger
        }
        style.color = palette.type === LIGHT ? WHITE : BLACK
      } else {
        children = name
      }
    } else {
      children = Icon? <Icon className={classes.fallback} />: <Person className={classes.fallback} />;
    }
    const cls = custom ? '' : clsx(
      classes.root,
      classes.system,
      classes[borderCls],
      classes[shape],
      classes[size],
      {
        [classes.colorDefault]: !!!src,
      },
      className,
    )
    const child = loadedSrc ? children : (
      <div 
        className="inline-flex  justify-center items-center"  style={cStyle}>
        {children}
      </div>)
    return (
      <Component
        data-test="table-avatar"
        className={cls}
        style={style}
        {...other}
        onClick={(e:any)=>handleClick&&handleClick(e)}
      >
       {child}
      </Component>
    );
}
export default withStyles(styles)(Avatar)