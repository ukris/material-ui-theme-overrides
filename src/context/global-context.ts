import { ThemeZoom }     from 'theme'
import { Buffer, FontSize } from 'types'
import  defaults, { Settings } from '../settings'
import { Cell, CellValue, Column } from 'apps/table/types'

const FULL_SCREEN = 'full_screen'
const THEME_ZOOM = 'theme_zoom'
const SETTINGS = 'settings'
const CURRENT_BUFFER = 'buffer'
const BUFFERS = 'buffers'
class LocalStorageMock {
    store:any={}
    constructor() {
      this.store = {};
    }
  
    clear() {
      this.store = {};
    }
  
    getItem(key:string) {
      return this.store[key] || null;
    }
  
    setItem(key:string, value:string) {
      this.store[key] = value.toString();
    }
  
    removeItem(key:string) {
      delete this.store[key];
    }
  };
  if (!global.localStorage) {
      // @ts-ignore
      global.localStorage = new LocalStorageMock()
  }
interface IGlobalContext {
    buffers: Buffer[]
    nextDomainId: number
    nextProjectId: number
    nextBoardId: number
    nextLaneId: number
    settings: Settings
    fullScreen: boolean
    palette?: any
    themeZoom: ThemeZoom,
    fontSize: FontSize,
    setThemeZoom: (themeZoom: ThemeZoom) => void
    setSettings : (settings: Settings) => void
    setFullScreen: (isFullScreen: boolean) => void
    focusedCell?: Cell | null
    currentBuffer?: Buffer
    domainId?: number
    projectId?: number
    boardId?: number
    laneBy?: number
    laneId?: string
    rows: any[]
    columns: Column[]
    anchorEl?: any
    setCurrentBuffer: (buffers: Buffer) => void
    setBuffers: (buffers: Buffer[]) => void
    setDomainId:(domainId: number) => void
    setProjectId:(projectId: number) => void
    setBoardId:(boardId: number) => void
    setLaneId:(laneId: string) => void
    setLaneBy:(labeBy: number) => void
    setRows: (rows:any[]) => void
    setColumns: (columns: Column[]) => void
    getColumn: (idx: number) => Column
    updateCell: (cell: Cell, newVal: CellValue) => void
    setFocusedCell:  (cell: Cell|null) =>void
    setAnchorEl: (el:any) => void
    theme: any
    setTheme: (theme:any)  => void
    /*

    setDomainId: (domainId: number) => void
    setProjectId: (projectId: number) => void
    setBoardId: (boardId: number) => void
    setLaneBy?: (laneBy?: string) => void // split by/ group by colId
    setLaneId: (laneId: string) => void // example  laneby status, laneId = 'FINISHED, 'INPROGRESS' etc 
    setRowIndex: (idx: number) => void
    setColIndex: (idx: number) => void
    setRowId: (id: string) => void
    setColId: (id: string) => void
    setRange: (range: number[]) => void
    */
}

const fullScreen = !!(localStorage && localStorage.getItem(FULL_SCREEN))
const tZoom = localStorage && localStorage.getItem(THEME_ZOOM)
const themeZoom:ThemeZoom = tZoom ? JSON.parse(tZoom) as ThemeZoom : { zoom: 1, theme: 'material'}
const defaultS = localStorage && localStorage.getItem(SETTINGS)
const defaultSettings: Settings = defaultS ? JSON.parse(defaultS) : defaults
let lBuffer = localStorage && localStorage.getItem(CURRENT_BUFFER)
const currentBuffer:Buffer = lBuffer ? JSON.parse(lBuffer) : []
lBuffer = localStorage && localStorage.getItem(BUFFERS)
const buffers:Buffer[] = lBuffer ? JSON.parse(lBuffer) : []

export const GlobalContext: IGlobalContext = {
    buffers,
    currentBuffer,
    nextDomainId: 2,
    nextProjectId: 2,
    nextBoardId: 2,
    nextLaneId: 2,
    fullScreen,
    themeZoom,
    settings: defaultSettings,
    theme: null,
    fontSize: {
        xs: '0.5rem',
        sm: '0.75rem',
        md: '1rem',
        lg: '1.25rem',
        xl: '1.5rem'
    },
    columns: [],
    rows: [],
    setFocusedCell: (cell: Cell|null)  => {
        GlobalContext.focusedCell = cell;
    },
    setThemeZoom: (themeZoom:ThemeZoom) => {
        (async() => localStorage && localStorage.setItem(THEME_ZOOM, 
            JSON.stringify(themeZoom)))()
        GlobalContext.themeZoom = themeZoom
    },
    setSettings: (settings: Settings) => {
        (async() => localStorage  && localStorage.setItem(SETTINGS, 
            JSON.stringify(settings)))()
        GlobalContext.settings = settings
    },
    setCurrentBuffer: (buffer: Buffer) => {
        (async() => localStorage  && localStorage.setItem(CURRENT_BUFFER, 
            JSON.stringify(buffer)))()
        GlobalContext.currentBuffer = buffer
    },
    setBuffers: (buffers: Buffer[]) => {
        (async() => localStorage  && localStorage.setItem(BUFFERS, 
            JSON.stringify(buffers)))()
        GlobalContext.buffers = buffers
    },
    setFullScreen: (screen: boolean) => {
        (async() => localStorage  && localStorage.setItem(FULL_SCREEN, screen.toString()))()
        GlobalContext.fullScreen = screen
    },
    setDomainId: (domainId: number) => {
        GlobalContext.domainId = domainId
    },
    setProjectId: (projectId: number) => {
        GlobalContext.projectId = projectId
    },
    setBoardId: (boardId: number) => {
        GlobalContext.boardId = boardId
    },
    setLaneId:(laneId: string) => {
        GlobalContext.laneId = laneId
    },
    setLaneBy:(laneBy: number) => {
        GlobalContext.laneBy = laneBy
    },
    setRows: (rows: any[]) => {
        GlobalContext.rows = rows
    },
    setColumns: (columns: Column[]) => {
        GlobalContext.columns = columns
    },
    getColumn: (idx:number) => GlobalContext.columns[idx],
    updateCell: (cell: Cell, newVal: CellValue) => {
        const { rowIndex, colIndex } = cell
        GlobalContext.rows[rowIndex][colIndex] = newVal
    },
    setAnchorEl: (el:any) => {
        GlobalContext.anchorEl = el
    },
    setTheme: (theme: any) => {
        GlobalContext.theme = theme
    }
}

export default GlobalContext