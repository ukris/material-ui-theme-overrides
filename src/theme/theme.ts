import { createMuiTheme, responsiveFontSizes} from '@material-ui/core'
import PaletteOptions from '@material-ui/core/styles/createMuiTheme';
import { GlobalContext } from 'context/global-context'
import { FONT_BASIS, makePalette, ThemeZoom, unit, tablePadding } from './index'
import { roundNum } from 'helpers'
// https://www.sipios.com/blog-tech/how-to-use-styled-components-with-material-ui-in-a-react-app
import typography from './typography'

function createTheme(variant: ThemeZoom):any {
  let palette:any = makePalette(variant)
  const { zoom } = variant
  const { colors } = palette
  delete palette.colors
  GlobalContext.palette = colors
  GlobalContext.palette.type = palette.type
  GlobalContext.palette.contrastText = palette.contrastText
  palette = {...palette, ...colors}
  GlobalContext.setThemeZoom(variant)
  const theme = createMuiTheme({
    zoomFontSize: (factor:number):string => `${roundNum(0.30 * factor * zoom)}${unit}`,
    zoomSpacing: (factor: number) => spacing(roundNum(factor * zoom)),
    fontSize: GlobalContext.fontSize,
    spacing: factor => `${roundNum(0.60 * factor)}${unit}`,
    lineHeight: GlobalContext.fontSize
  });
  const { spacing, zoomSpacing, zoomFontSize } = theme
  const fontSize =  {
      xs: zoomFontSize(FONT_BASIS),
      sm: zoomFontSize(FONT_BASIS+1),
      md: zoomFontSize(FONT_BASIS+2),
      lg: zoomFontSize(FONT_BASIS+3),
      xl: zoomFontSize(FONT_BASIS+4)
  }
  const lineHeight =  {
    xs: zoomFontSize(FONT_BASIS+0.5),
    sm: zoomFontSize(FONT_BASIS+1.5),
    md: zoomFontSize(FONT_BASIS+2.5),
    lg: zoomFontSize(FONT_BASIS+3.5),
    xl: zoomFontSize(FONT_BASIS+4.5)
  }
  let padding:string = tablePadding[0].toString()
  if (tablePadding[0]) {
    padding+= unit
  }
  padding+= ' '
  padding+= tablePadding[1].toString()
  if (tablePadding[1]) {
    padding+= unit
  }
  GlobalContext.fontSize = fontSize
  const modTheme:any = createMuiTheme(
    /**
     * @see https://material-ui.com/customization/themes/#theme-configuration-variables
     * https://material-ui.com/customization/typography/
     */
    {
      palette,
      zoomFontSize,
      typography,
      spacing,
      zoomSpacing,
      fontSize,
      lineHeight,
      props: {
        MuiButtonBase: {
          disableRipple: true
        },
        MuiButton: {
            variant: 'contained',
            disableElevation: true,
        },
        MuiTextField: {
            variant: 'outlined'
        }
      },
      zIndex: {
        appBar: 1200,
        drawer: 1100,
      },
      overrides: {
        MuiDrawer: {
          paper: {
            background: '#18202c',
          },
        },
        MuiPaper: {
          // Name of the component ⚛️ / style sheet
          elevation0: {
            // Name of the rule
            boxShadow: '0 0 14px 0 rgba(53,64,82,.05)', // Some CSS
          },
        },
        MuiTableCell: {
          root: { 
            borderBottom: '1px solid rgba(224, 224, 224, .5)', 
            lineHeight: lineHeight.md,
            fontSize: fontSize.md,
            backgroundColor: theme.palette.background.default,
          },
        },
        MuiAutocomplete: {
          root: {
            backgroundColor: theme.palette.background.default,
            color: theme.palette.getContrastText(theme.palette.background.default)
          }
        },
        MuiButton: {
          root: {
            color: theme.palette.primary.contrastText,
          }
        },
        MuiTextField: {
          root: {
            variant: 'outlined'
          }
        }
    }}
  )
  //  return responsiveFontSizes(modTheme)
  GlobalContext.theme = theme
  return modTheme
}

export default createTheme