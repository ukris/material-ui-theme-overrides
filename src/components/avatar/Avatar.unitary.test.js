import React from 'react'
import { GlobalContext } from 'context'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Avatar from './Avatar'
import { ThemeContext, ThemeProvider } from 'theme'

configure({ adapter: new Adapter() })

const themeZoom = GlobalContext.themeZoom;
const setThemeZoom = () => true

describe('Avatar Component | Valid Image', () => {
    const component = (
      <ThemeContext.Provider value={{ themeZoom, setThemeZoom }}>
        <ThemeProvider>
            <Avatar
                src="https://material-ui.com/static/images/avatar/2.jpg"
                name="Genzo Wonzo"
                size="xs"
                shape="square"
            />
        </ThemeProvider>
      </ThemeContext.Provider>
    )

    const wrapper = mount(component)

    it('renders component correctly', () => {
        const avatar = wrapper.props().children.props
        
        expect(avatar.src).toEqual("https://material-ui.com/static/images/avatar/2.jpg")
        expect(avatar.name).toEqual("Genzo Wonzo")
        expect(avatar.size).toEqual("xs")
        expect(avatar.shape).toEqual("square")
        expect(avatar.border).toEqual(undefined)
    });

    it('renders initials', () => {
        const img = wrapper.find('[data-test="table-avatar"]')

        expect(img.length).toEqual(1)
        expect(img.props().children).toEqual('GW')
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
    });
})

describe('Avatar Component | Invalid Image', () => {
    const component = (
      <ThemeContext.Provider value={{ themeZoom, setThemeZoom }}>
        <ThemeProvider>
            <Avatar
                src="https://www.gravatarz.com/avatar/fooff"
                name="Natasha Special"
                size="xl"
                shape="round"
                border
            />
        </ThemeProvider>
      </ThemeContext.Provider>
    )

    const wrapper = mount(component)

    it('renders component correctly', () => {
        const avatar = wrapper.props().children.props
        
        expect(avatar.src).toEqual("https://www.gravatarz.com/avatar/fooff")
        expect(avatar.name).toEqual("Natasha Special")
        expect(avatar.size).toEqual("xl")
        expect(avatar.shape).toEqual("round")
        expect(avatar.border).toEqual(true)
    });

    it('renders initials', () => {
        const img = wrapper.find('[data-test="table-avatar"]')

        expect(img.length).toEqual(1)
        expect(img.props().children).toEqual('NS')
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
    });
})

describe('Avatar Component | No Image', () => {
    const component = (
      <ThemeContext.Provider value={{ themeZoom, setThemeZoom }}>
        <ThemeProvider>
            <Avatar />
        </ThemeProvider>
      </ThemeContext.Provider>
    )

    it('renders fallback', () => {
        const wrapper = mount(component)
        const img = wrapper.find('[data-test="table-avatar"]')

        expect(img.length).toEqual(1)
        expect(img.props().children.props.className.includes('Avatar-fallback')).toBe(true)
        expect(wrapper.find('svg').length).toEqual(1)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
    });
})