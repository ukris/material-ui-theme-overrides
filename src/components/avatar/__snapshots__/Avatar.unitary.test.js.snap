// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component | Invalid Image matches snapshot 1`] = `
Array [
  " ",
  <span
    className="Avatar-root-43 Avatar-border-46 Avatar-round-48 Avatar-xl-56"
    data-test="table-avatar"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#707264",
        "color": "#fff",
      }
    }
  >
    NS
  </span>,
]
`;

exports[`Avatar Component | No Image matches snapshot 1`] = `
Array [
  " ",
  <span
    className="Avatar-root-80 Avatar-noClass-84 Avatar-circle-82 Avatar-md-91 Avatar-colorDefault-81"
    data-test="table-avatar"
    onClick={[Function]}
    style={Object {}}
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root Avatar-fallback-88"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </span>,
]
`;

exports[`Avatar Component | Valid Image matches snapshot 1`] = `
Array [
  " ",
  <span
    className="Avatar-root-29 Avatar-noClass-33 Avatar-square-35 Avatar-xs-38"
    data-test="table-avatar"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#8a564e",
        "color": "#fff",
      }
    }
  >
    GW
  </span>,
]
`;
